lib/matplotlib/pyplot.py

```diff
@@ -1845,27 +1850,19 @@ def get_plot_commands():
     """
     Get a sorted list of all of the plotting commands.
     """
-    # This works by searching for all functions in this module and
-    # removing a few hard-coded exclusions, as well as all of the
-    # colormap-setting functions, and anything marked as private with
-    # a preceding underscore.
-
-    import inspect
-
+    # This works by searching for all functions in this module and removing
+    # a few hard-coded exclusions, as well as all of the colormap-setting
+    # functions, and anything marked as private with a preceding underscore.
     exclude = {'colormaps', 'colors', 'connect', 'disconnect',
                'get_plot_commands', 'get_current_fig_manager', 'ginput',
                'plotting', 'waitforbuttonpress'}
     exclude |= set(colormaps())
     this_module = inspect.getmodule(get_plot_commands)
-
-    commands = set()
-    for name, obj in list(six.iteritems(globals())):
-        if name.startswith('_') or name in exclude:
-            continue
-        if inspect.isfunction(obj) and inspect.getmodule(obj) is this_module:
-            commands.add(name)
-
-    return sorted(commands)
+    return sorted(
+        name for name, obj in globals().items()
+        if not name.startswith('_') and name not in exclude
+           and inspect.isfunction(obj)
+           and inspect.getmodule(obj) is this_module)
```

Comment:
Side note: as an argument to `sorted`, using a list comprehension is faster than using a bare generator:
```
In [10]: timeit sorted([i for i in range(100)])
7.52 µs ± 119 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [11]: timeit sorted(i for i in range(100))
10.1 µs ± 223 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```
Of course, it doesn't matter unless one is dealing with a large number of items or calls.

Reply:
Actually I'm puzzled how that can be the case.  https://bugs.python.org/issue32945